full/src/test/java/apoc/load/LoadHtmlTest.java

```diff
@@ -122,4 +125,65 @@ public void testQueryWithChildren() {
                     assertEquals(1, ((List) firstChild.get("children")).size());
                 });
     }
+
+    @Test
+    public void testQueryWithFailsSilently() {
+        Map<String, Object> query = map("h2", "h2","a", "a", "invalid", "invalid");
+
+        testResult(db, "CALL apoc.load.html($url,$query, {failSilently: true})",
+                map("url", new File("src/test/resources/wikipedia.html").toURI().toString(), "query", query),
+                result -> {
+                    Map<String, Object> row = result.next();
+                    Map<String, Object> value = (Map<String, Object>) row.get("value");
+                    // number of <a> tags in html file minus the incorrect tag
+                    assertEquals(107, ((List) value.get("a")).size());
+                    assertEquals(Collections.emptyList(), value.get("invalid"));
+                    assertEquals(List.of(RESULT_QUERY_H2).toString().trim(), value.get("h2").toString().trim());
+                    assertFalse(result.hasNext());
+                });
+    }
+
+    @Test(expected = QueryExecutionException.class)
+    public void testQueryWithoutFailsSilently() {
+        final String url = new File("src/test/resources/wikipedia.html").toURI().toString();
+        try {
+            Map<String, Object> query = map("a", "a", "h2", "h2");
+            testCall(db, "CALL apoc.load.html($url,$query)", map("url", url, "query", query), (r) -> {});
+        } catch (Exception e) {
+            Throwable except = ExceptionUtils.getRootCause(e);
+            String expectedMessage = "Error during parsing element: <a hre f=\"#cite_ref-Simpson2018_1-0\"><sup><i><b>a</b></i></sup></a>";
+            assertEquals(expectedMessage, except.getMessage());
+            throw e;
+        }
+    }
+
+    @Test(expected = QueryExecutionException.class)
+    public void testQueryWithExceptionIfIncorrectUrl() {
+
+        final String filePath = new File("src/test/resources/wikipedia1.html").toURI().toString();
+        try {
+            String query = "CALL apoc.load.html('" + filePath + "',{a:'a'})";
```

Comment:
might be worth adding the variant with `failSilently: true` (just to "prove" it does not have an effect on unrecoverable errors)
same comment for the other test below